### ISSUE TITLE
SpatialDecon using readGeoMx

### ISSUE THREAD

**Issue by jjoo12** [DavisLaboratory/standR#31]
Hello,

I am a student currently analyzing GeoMx data using StandR. I am in the process of learning spatial analysis and encountered a specific issue, so I decided to reach out on GitHub.

When creating spatial experiment data using the readGeoMx function, there were no issues following the default flow, where rmNegProbe is carried out. However, when attempting spatial deconvolution with rmNegProbe set to FALSE, I am encountering a duplicate error. I suspect this might be related to the usage of WTA data.(There are 139 NegProbe-WTX) How is this typically resolved in such situations?

Thank you!

This is the error code

Error in `.rowNamesDF<-`(x, value = value) : 
  duplicate 'row.names' are not allowed
In addition: Warning message:
non-unique value when setting 'row.names': ‘NegProbe-WTX’ 

Thank you

**Comment by hookoop** [comment]
We are getting the following error: 
 "NegProbe-WTXmust be included in the dataset. Perhaps specify rmNegProbe=TRUE when using readGeoMx" 
when using `prepareSpatialDecon`.
We are using WTA so there are several negative probes, however we do not understand what the `pool=` variable in the command should look like. The vignette says "A vector indicates the pools of the genes.", but what does this mean? A vector of gene names?

Best,
Helka

**Comment by ningbioinfo** [comment]
hi @hookoop , the input for the `pool` argument is a vector with length of the genes, i.e. nrow(spe). It should be something like c("negprob1","negprob1","negprob1","negprob2","negprob2","negprob2"..., "negprobx"), so it tells the function with negative probes is governing which genes.

**Comment by ningbioinfo** [comment]
hi @jjoo12 , what you need to do is first change the name of the negprobe-wtx into maybe negprobe-wtx1,negprobe-wtx2,negprobe-wtx3,..., negprobe-wtx139. Then use the `pool` argument in the function to specify which negprobe is responsible for which gene

**Comment by hookoop** [comment]
Hey @ningbioinfo , how do we find out which negative probe is assigned to which gene? We couldn't figure this out based off the file we got from GeoMX data analysis suite

**Comment by ningbioinfo** [comment]
Hi @hookoop , I think that's a question for people who run the experiment then.

**Comment by hookoop** [comment]
Hi @ningbioinfo , we ran the experiment ourselves. 
@jjoo12 were you able to get `prepareSpatialDecon` to work with the WTA?

**Comment by ningbioinfo** [comment]
Hi @hookoop , I see, it should be related to the probe designed etc. I wrote this function based on what they required for the `spatialDecon` package, which is the tool from the Nanostring company, perhaps send through a question to the company for enquiry the relevant information?

**Comment by Anjaney10** [comment]
Hi, was anyone able to solve this? I am struggling with the same issue. I have 138 NegProbe-WTX, and I don't know which NegProbe is responsible for which gene. What should we do in this case? Any suggestions will be appreciated!

**Comment by hookoop** [comment]
I talked with the tech support about this 1,5 years ago and they said that there are no specific negative probes for specific genes, but the mean of the bakcground probes is used to estime bakcground for a given segment. If you only used one probe set, like the Human transcriptome atlas, you shoudl be setting rmNegProbe=FALSE. This in turn will lead to a "duplicate rownames" issue. I think we solved this by giving the negativeprobe unique names (NegProbeWTX-01, -02, -03, .. -139).
